server/paymentProviders/stripe/webhook.ts

```diff
@@ -160,7 +160,9 @@ export const paymentIntentSucceeded = async (event: Stripe.Event) => {
   const paymentIntent = event.data.object as Stripe.PaymentIntent;
   const order = await models.Order.findOne({
     where: {
-      status: OrderStatuses.PROCESSING,
+      // Since new CreditCard donations are not synchronously validated, we also need to consider those orders that
+      // go straight from NEW to PAID with payment_intent.succeeded.
+      status: [OrderStatuses.NEW, OrderStatuses.PROCESSING, OrderStatuses.ERROR, OrderStatuses.ACTIVE],
```

Comment:
Why do we even filter on limited list of status? Should we consider not having a filter, or exclude instead?

Reply:
I think it is a good practice to be explicit about what we're waiting for. If we receive an event for an order that is already marked as PAID, that is a problem that we would need to know.

Reply:
We're accepting `ACTIVE` here, which is not far to be like `PAID`.

Reply:
True. I'll drop it since we're now clearing data.paymentIntent after processing.